### PR TITLE
Set IsDynamicCodeCompiled as a managed constant to help

### PR DESCRIPTION
**Pull request by marek-safar**
with trimming on platforms where it's always false

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
Is there a rule for when we use substitution and when ifdef (`IsDynamicCodeCompiled` is a substitution)?

**Comment by marek-safar** [comment]
For me substitutions are more suitable for logic that is optional

**Comment by MichalStrehovsky** [comment]
Is the substitution for this still needed then?

https://github.com/dotnet/runtime/blob/75476f9eedc35dbe8e65bba851e7a1fca851d7ca/src/mono/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.iOS.xml#L1-L7

https://github.com/dotnet/runtime/blob/75476f9eedc35dbe8e65bba851e7a1fca851d7ca/src/mono/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.wasm.xml#L9-L11

File: src/mono/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeFeature.Mono.cs [hunk 1]
```diff
@@ -14,8 +14,12 @@ public static bool IsDynamicCodeSupported
 
         public static bool IsDynamicCodeCompiled
         {
+#if TARGET_BROWSER || TARGET_IOS || TARGET_TVOS
```
Comment (lambdageek) [inline src/mono/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeFeature.Mono.cs:17]:
```suggestion
#if TARGET_BROWSER || TARGET_IOS || TARGET_TVOS || TARGET_MACCATALYST
```

**Review by lambdageek — APPROVED** [review]

**Comment by steveisok** [comment]
@marek-safar do you think we still needs this?

**Comment by eerhardt** [comment]
Closing, as this PR is stale and conflicts with #80246.

One comment I have here is that we run the ILLinker on CoreLib during dotnet/runtime's build. So the substitutions are made into the IL in the shipped assemblies already. I'm not sure there is an externally visible difference with this change.

**Pull request closed without merging**